File: front/kaemeweb/src/app/modules/shared/services/tipo-peca.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { Cliente } from '../models/cliente';
import { Fornecedor } from '../models/fornecedor';
import { TipoPeca } from '../models/tipo-peca';

@Injectable({
  providedIn: 'root'
})
export class TipoPecaService {

  constructor(private httpClient: HttpClient) { }

  create(tipo_peca: TipoPeca): Observable<boolean> {
    return this.httpClient.post<boolean>('https://localhost:7072' + '/tipopeca/Create', tipo_peca);
  }

  update(tipo_peca: TipoPeca): Observable<boolean> {
    return this.httpClient.put<boolean>('https://localhost:7072' + '/tipopeca/Update', tipo_peca);
  }

  delete(id: number): Observable<boolean> {
    let params = new HttpParams()
    params = params.append('id', id)

    return this.httpClient.delete<boolean>('https://localhost:7072' + '/tipopeca/Delete', {params});
  }

  getAll(): Observable<TipoPeca[]> {
    return this.httpClient.get<TipoPeca[]>('https://localhost:7072' + '/tipopeca/GetAll');
  }

  getAllAtivos(): Observable<TipoPeca[]> {
    return this.httpClient.get<TipoPeca[]>('https://localhost:7072' + '/tipopeca/GetAllAtivos');
  }

}

File: back/KaemeWebApi/Dominio/Services/TipoPecaService.cs
using Dominio.Interface;
using Dominio.Models;
using Shared.RepositoryShared;

namespace Dominio.Services
{
    public class TipoPecaService
    {
        private ITipoPecaRepository _tipoPecaRepository;

        public TipoPecaService(ITipoPecaRepository tipoPecaRepository)
        {
            _tipoPecaRepository = tipoPecaRepository;
        }

        public RepositoryResult Create(TipoPeca tipoPeca)
        {
            try
            {
                var result = _tipoPecaRepository.Create(tipoPeca);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult Update(TipoPeca tipoPeca)
        {
            try
            {
                var result = _tipoPecaRepository.Update(tipoPeca);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }        

        public RepositoryResult Delete(int id)
        {
            try
            {
                var result = _tipoPecaRepository.Delete(id);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }         

        public RepositoryResult GetAll()
        {
            try
            {
                var result = _tipoPecaRepository.GetAll();
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetAllAtivos()
        {
            try
            {
                var result = _tipoPecaRepository.GetAllAtivos();
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }        

    }
}